opencog/atoms/execution/GetMethodLink.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * opencog/atoms/execution/GetMethodLink.cc
+ *
+ * Copyright (C) 2019 OpenCog Foundation
+ * All Rights Reserved
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <opencog/atoms/execution/GroundedObjectNode.h>
+
+#include "GetMethodLink.h"
+
+using namespace opencog;
+
+GetMethodLink::GetMethodLink(const HandleSeq& output_set, Type type)
+	: GroundedFunctionLink(output_set, type)
+{
+	check_outgoing_type(0, GROUNDED_OBJECT_NODE);
+	check_outgoing_type(1, CONCEPT_NODE);
```

Comment:
Why are you checking here as opposed to a validator (I don't have an opinion on the matter, just curious)?

Reply:
Actually it is a question for me which way to use. Most of atomspace code uses checks in constructors, like recent `CondLink`. And validators are used for three types only.

BTW it is a place where I have enchanced atomspace outside of opencog repository, I mean I have added `OrderedLink` class to implement arguments check.

Reply:
Not sure `OrderedLink` is needed actually, will look at alternative validator implementation to not introduce more inconsistencies then it is needed.

Reply:
Actually there is not standard way to define Validator near the Atom which uses the validator, you need to add it into Checkers.cc which is ugly. So I am going to make a fix in the opencog repo. And I just will move all of the code into GetMethodLink.cc before that.

Reply:
I agree that it's ugly. If you know how to beautify it, it's welcome :-)